modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCFeatureSource.java

```diff
@@ -908,5 +913,145 @@ private boolean handleNearestVisitor(Query query, FeatureVisitor visitor) throws
         
         return result;
     }
-    
+
+    @Override
+    protected List<String> getNativeStyles() {
+        List<String> list = new ArrayList<>();
+        String featureName = this.getName().toString();
+        Boolean layerstylesexist = false;
+        try {
+            JDBCState state = getState();
+            Connection cx = getDataStore().getConnection(state);
+            DatabaseMetaData metaData = cx.getMetaData();
+            String[] queryTypes = {"TABLE"};
+            ResultSet tables = metaData.getTables(null, null, "layer_styles", queryTypes);
+            if (tables.next()) {
+                layerstylesexist = true;
+            }
+            tables.close();
+            if (layerstylesexist) {
+                Statement st = cx.createStatement();
+                String sql = null;
+                String styleNameColumnName = null;
+                // GeoPackage DataSource class org.sqlite.javax.SQLiteConnectionPoolDataSource
+                // PostGIS    DataSource class org.geotools.data.jdbc.datasource.DBCPDataSource
+                if (getDataStore().getSQLDialect() instanceof PreparedStatementSQLDialect) {
+                    // GeoPackage
+                    sql = "select * from layer_styles where f_table_name = '" + featureName + "'";
+                    styleNameColumnName = "styleName";
+                } else {
+                    // Postgresql
+                    sql = "select * from public.layer_styles where f_table_name = '" + featureName + "'";
```

Comment:
This query should be agnostic to the pg schema the the data comes from, not limited to the public schema

Reply:
@mprins Thank you for your comment. You pointed out a very important issue. How to we store styles on the provider? In this [proposal](https://github.com/geotools/geotools/wiki/ResourceInfo-with-native-style), we are just following the QGIS approach to store the style on a single table named `layer_styles` on the `public` schema.

One of the arguments for this PR is to enhance the interoperability between GeoTools/GeoServer with QGIS.

We can make a better proposal to store native styles for GeoPackage and PostGIS and then improve both GeoTools and QGIS to support it.

Reply:
Agree with @mprins this is potentially a problem, particularly in this place, as it's shared among all database implementations (do Oracle and SQLServer even have a "public" schema?).
Maybe it's best moved to SQLDialect, and implemented only for postgis and geopackage.
For postgis at least, I can see the usage of public schema as "valid", geometry_columns is in there after all. Looking at the implementation however, I see we don't schema qualify the lookup, allowing the search path to drive the search. Maybe this could be done for the postgis implementation of the style lookup too?